integration/components/image/image_test.go

```diff
@@ -0,0 +1,156 @@
+// limitations under the License.
+
+package image
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/spf13/cobra"
+	"github.com/stretchr/testify/require"
+
+	commonImage "github.com/inspektor-gadget/inspektor-gadget/cmd/common/image"
+	. "github.com/inspektor-gadget/inspektor-gadget/integration"
+)
+
+func TestImage(t *testing.T) {
+	// ensure that experimental is enabled
+	os.Setenv("IG_EXPERIMENTAL", "true")
+
+	// start registry
+	r := StartRegistry(t, "test-image-registry")
+	t.Cleanup(func() {
+		r.Stop(t)
+	})
+	if len(r.PortBindings()) == 0 || r.PortBindings()["5000/tcp"] == nil {
+		t.Fatal("registry port not exposed")
+	}
+	pb := r.PortBindings()["5000/tcp"][0]
+	registryAddr := fmt.Sprintf("%s:%s", pb.HostIP, pb.HostPort)
+
+	testReg := "reg.com"
+	testRepo := "repo1"
+	testTag := "tag1"
+	testLocalImage := fmt.Sprintf("%s/%s:%s", testReg, testRepo, testTag)
+	testRegistryImage := fmt.Sprintf("%s/%s:%s", registryAddr, testRepo, testTag)
+
+	// ensure all images are removed
+	t.Cleanup(func() {
+		// remove local image
+		cmd := commonImage.NewRemoveCmd()
+		cmd.SetArgs([]string{testLocalImage})
+		cmd.Execute()
+
+		// remove registry image
+		cmd = commonImage.NewRemoveCmd()
+		cmd.SetArgs([]string{testRegistryImage})
+		cmd.Execute()
+	})
+
+	type testCase struct {
+		name           string
+		cmd            *cobra.Command
+		args           []string
+		expectedStdout []string
+	}
+
+	testCases := []testCase{
+		{
+			name: "build",
+			cmd:  commonImage.NewBuildCmd(),
+			args: []string{
+				"--builder-image", *testBuilderImage, "--tag", testLocalImage, "../../../gadgets/trace_open",
+			},
+			expectedStdout: []string{
+				fmt.Sprintf("Successfully built %s", testLocalImage),
+			},
+		},
+		{
+			name: "list",
+			cmd:  commonImage.NewListCmd(),
+			args: []string{},
+			expectedStdout: []string{
+				testRepo,
+				testTag,
+			},
+		},
+		{
+			name: "tag",
+			cmd:  commonImage.NewTagCmd(),
+			args: []string{testLocalImage, testRegistryImage},
+			expectedStdout: []string{
+				fmt.Sprintf("Successfully tagged with %s", testRegistryImage),
+			},
+		},
+		{
+			name: "push",
+			cmd:  commonImage.NewPushCmd(),
+			args: []string{testRegistryImage, "--insecure"},
+			expectedStdout: []string{
+				fmt.Sprintf("Successfully pushed %s", testRegistryImage),
+			},
+		},
+		{
+			name: "remove-local",
+			cmd:  commonImage.NewRemoveCmd(),
+			args: []string{testLocalImage},
+			expectedStdout: []string{
+				fmt.Sprintf("Successfully removed %s", testLocalImage),
+			},
+		},
+		{
+			name: "remove-registry",
+			cmd:  commonImage.NewRemoveCmd(),
+			args: []string{testRegistryImage},
+			expectedStdout: []string{
+				fmt.Sprintf("Successfully removed %s", testRegistryImage),
+			},
+		},
+		{
+			name: "pull",
+			cmd:  commonImage.NewPullCmd(),
+			args: []string{testRegistryImage, "--insecure"},
+			expectedStdout: []string{
+				fmt.Sprintf("Successfully pulled %s", testRegistryImage),
```

Comment:
Could you add other `NewPullCmd` entries in `testCases` to test the following:
- pulling the same gadget twice would say it is already pulled
- pulling a non-existent gadget would return an error such as `failed to resolve`, `connection refused` or `does not exist`.

Reply:
> pulling the same gadget twice would say it is already pulled

It seems we don't print any special message like `Image is up to date for ...`  if image is pulled twice, so something to handle in another PR? 

```
→ sudo -E ./ig image pull ghcr.io/inspektor-gadget/gadget/trace_open:v0.24.0
INFO[0000] Experimental features enabled                
Pulling ghcr.io/inspektor-gadget/gadget/trace_open:v0.24.0...
Successfully pulled ghcr.io/inspektor-gadget/gadget/trace_open:v0.24.0@sha256:8c920ff77d48caf9c0012fa61e32cbffb73a285c6d652ac9703d31d7f5d40876
→ sudo -E ./ig image pull ghcr.io/inspektor-gadget/gadget/trace_open:v0.24.0
INFO[0000] Experimental features enabled                
Pulling ghcr.io/inspektor-gadget/gadget/trace_open:v0.24.0...
Successfully pulled ghcr.io/inspektor-gadget/gadget/trace_open:v0.24.0@sha256:8c920ff77d48caf9c0012fa61e32cbffb73a285c6d652ac9703d31d7f5d40876

```

> pulling a non-existent gadget would return an error such as failed to resolve, connection refused or does not exist.

Great idea with negative tests. I added some! :)